crates/ruff_linter/src/linter.rs

```diff
@@ -521,16 +586,39 @@ pub fn lint_fix<'a>(
             }
         }
 
+        #[cfg(not(feature = "test-rules"))]
+        let diagnostics = &result.data.0;
+
+        // HACK(zanieb): We cannot raise violations for "fixed" test diagnostics
+        //               or we will never converge since they raise unconditionally
+        //               so we filter them out here
+        #[cfg(feature = "test-rules")]
+        let filtered_diagnostics = result
+            .data
+            .0
+            .iter()
+            .filter(|diagnostic| !applied_test_fixes.contains(&diagnostic.kind.rule()))
+            .cloned()
+            .collect::<Vec<_>>();
+        #[cfg(feature = "test-rules")]
+        let diagnostics = filtered_diagnostics.as_slice();
```

Comment:
This is ugly and I do not think it is working correctly — the snapshots are limited to a single fix now but they include diagnostics for the "fixed" test rules

Reply:
(Because we do not filter these downstream in `result.map`)